### PR TITLE
Feed management

### PR DESCRIPTION
**Pull request by cvankerkhove**
This pull request contains development of the feed management module, paticulary from the feed inventory pseudo code. The following document attached provides a basic outline and descriptions of the main functions added and how they interact to help run the feed management system in RuFas. 
[RUFAS Feed Management Pull Request Outline.pdf](https://github.com/RuminantFarmSystems/MASM/files/4841396/RUFAS.Feed.Management.Pull.Request.Outline.pdf)

### CODE REVIEW

File: Inputs/animal_management.json [hunk 1]
```diff
@@ -475,16 +478,12 @@
...
-            "managed_feeds":
-            [
-              "Corn_grain",
-              "Legume_hay",
-              "Cotton_seed",
-              "Roasted_soybean",
-              "Rye_hay"
-            ],
+            "purchased_feeds": [2, 24, 34, 38, 91, 102, 137],
+            "growing_feeds": [8, 87],
```
Comment (tlh96) [inline Inputs/animal_management.json]: I assume these changes are needed with the feed library. In the output graphics, the numbers are showing, not feed names. 

Is there a way to add the feed names to the output graphic axes? @militsasotirova @wmaxdonovan

Comment (militsasotirova) [reply Inputs/animal_management.json]: I thought there could be a work-around that makes the display the feed names in the output, but I don't see a way to differentiate between the different possible qualities of the grown feeds at the very beginning when the csv headers are written (this brings up the question - do we want to treat each quality of a feed as separate in the output?).

Comment (militsasotirova) [reply Inputs/animal_management.json]: Fixed in update_feed_library

**Review by tlh96 — APPROVED** [review]
I reviewed changes in feed.py related to feed inventory management. Overall, the code is well commented and logical in order. 

Some minor formatting changes (line length) can be completed to meet python style conventions. Chris offered to update the pseudocode variable name changes to match the code.

I ran several scenarios with immature, mid-maturity, and mature forages. Functions work as anticipated. 

QUESTION @KFosterReed How do we account for more immature and mid-maturity forage than needed for lactating cows?

File: RUFAS/routines/feed/feed.py [hunk 2]
```diff
@@ -341,40 +377,294 @@ def summarize_feed_storage(self, storage):
...
+            inventory plan. If the list new_forages contains a forage which
+            had been harvested at least 30 days ago, forage_inv_plan will be called.
+
+        Args:
+            animal_management: The state of the AnimalManagement class object
+        """
+        # Daily feedout for silos with farm grown forages in them per pen based on ration formulated
+        for silo in self.storage_options:
+            if self.storage_options[silo].days_since_feedout >= 0 and self.storage_options[silo].DM > 0:
+                ration = hardcoded_ration.get_ration()
```
Comment (militsasotirova) [inline RUFAS/routines/feed/feed.py:637]: I would add a TODO here to change this once we get the animal rations in master

**Review by wmaxdonovan — APPROVED** [review]
Code, commenting, and functionality look good to me. Only significant change I made was to put the animal initialization behind a simulate_animal if statement to take time off simulations with herd size 0.

**Review by militsasotirova — APPROVED** [review]
Other than the TODO, I think everything looks good!

**Pull request merged**